src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs

```diff
@@ -97,24 +109,83 @@ public class MaturedBlocksSyncManager : IMaturedBlocksSyncManager
             this.initialBlockDownloadState = initialBlockDownloadState;
             this.nodeLifetime = nodeLifetime;
             this.conversionRequestRepository = conversionRequestRepository;
+            this.counterChainSettings = counterChainSettings;
+            this.httpClientFactory = httpClientFactory;
             this.chainIndexer = chainIndexer;
             this.externalApiPoller = externalApiPoller;
             this.conversionRequestFeeService = conversionRequestFeeService;
             this.network = network;
             this.federationManager = federationManager;
+            this.mainChainActivationHeight = int.MaxValue;
 
             this.lockObject = new object();
             this.logger = LogManager.GetCurrentClassLogger();
         }
 
+        public class ConsensusClient : RestApiClientBase
```

Comment:
We actually have a `FederationGatewayClient` already that has all the classes you need obn there already. Can you move the `GetLockedInDeployments` method there?

Reply:
The 'FederationGatewayClient` assumes that the controller is `FederationGatewayController`. The method I am using already exists on the `ConsensusController`.

Reply:
Class renamed and moved to its own file.